core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -202,62 +202,52 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
-    * where workers see only the elements after the start of each worker evaluation.
-    *
-    * Note: the resulting stream will not emit values, even if the pipes do.
-    * If you need to emit `Unit` values, consider using `broadcastThrough`.
+    * The pipes are all run concurrently with each other, but note
+    * that elements are pulled from the source as chunks, and the next
+    * chunk is pulled only when all pipes are done with processing the
+    * current chunk, which prevents faster pipes from getting too far ahead.
     *
-    * Note:  Elements are pulled as chunks from the source and the next chunk is pulled when all
-    * workers are done with processing the current chunk. This behaviour may slow down processing
-    * of incoming chunks by faster workers.
-    * If this is not desired, consider using the `prefetch` and `prefetchN` combinators on workers
-    * to compensate for slower workers.
-    *
-    * @param pipes    Pipes that will concurrently process the work.
-    */
-  def broadcastTo[F2[x] >: F[x]: Concurrent](
-      pipes: Pipe[F2, O, Nothing]*
-  ): Stream[F2, INothing] =
-    this.through(Broadcast.through(pipes: _*))
-
-  /** Variant of `broadcastTo` that broadcasts to `maxConcurrent` instances of a single pipe.
-    */
-  def broadcastTo[F2[x] >: F[x]: Concurrent](
-      maxConcurrent: Int
-  )(pipe: Pipe[F2, O, Nothing]): Stream[F2, INothing] =
-    this.broadcastTo[F2](List.fill(maxConcurrent)(pipe): _*)
-
-  /** Alias for `through(Broadcast.through(pipes))`.
+    * In other words, this behaviour slows down processing of incoming
+    * chunks by faster pipes until the slower ones have caught up. If
+    * this is not desired, consider using the `prefetch` and
+    * `prefetchN` combinators on the slow pipes.
     */
   def broadcastThrough[F2[x] >: F[x]: Concurrent, O2](
       pipes: Pipe[F2, O, O2]*
   ): Stream[F2, O2] =
-    through(Broadcast.through(pipes: _*))
-
-  /** Variant of `broadcastTo` that broadcasts to `maxConcurrent` instances of the supplied pipe.
-    */
-  def broadcastThrough[F2[x] >: F[x]: Concurrent, O2](
-      maxConcurrent: Int
-  )(pipe: Pipe[F2, O, O2]): Stream[F2, O2] =
-    this.broadcastThrough[F2, O2](List.fill(maxConcurrent)(pipe): _*)
+    Stream
+      .eval {
+        (
+          CountDownLatch[F2](pipes.length),
+          fs2.concurrent.Topic[F2, Option[Chunk[O]]]
+        ).tupled
+      }
+      .flatMap { case (latch, topic) =>
+        Stream(pipes: _*)
+          .map { pipe =>
+            Stream
+              .resource(topic.subscribeAwait(1))
+              .flatMap { sub =>
+                // crucial that awaiting on the latch is not passed to
+                // the pipe, so that the pipe cannot interrupt it and alter
+                // the latch count
+                Stream.exec(latch.release >> latch.await) ++
+                  sub.unNoneTerminate.flatMap(Stream.chunk).through(pipe)
+              }
```

Comment:
Could we extract this lambda block as a function `def pump(pipe:  Pipe[F2, O, O2])` ?

Reply:
I'm factored out the `unNoneTerminate...` stuff, but I want to keep the waiting logic all together, so it's easy to understand how to `parJoin`, the `latch` and the `Resource` all work